### ISSUE TITLE
Customized Id property

### ISSUE THREAD

**Issue by jmls** [gabrielguerrero/ngrx-traits#85]
ngrx/signalstore supports entities with custom id fields 

https://ngrx.io/guide/signals/signal-store/entity-management#customized-id-property

If I try and pass a type that doesn't have the `id` field defined, then I get a compile error with withEntitiesSingleSelection , withEntitiesMultiSelection and  withEntitiesLocalFilter saying

"not assignable to type '{ id: string | number; }'."

doess ngrx-traits alllow for custom id fields ?

**Comment by gabrielguerrero** [comment]
Not at the moment, but it can be added, will work on it soon should not be a big change

**Comment by gabrielguerrero** [comment]
Ok this is what I found, my original idea was to add a new optional prop idKey, to all withEntities* , and this will need to be passed on to each in the same way that collection is passed, there are a few problems with this approach, for one not all of the withEntities need to change the entities so they dont all need the idKey prop, and because is optional it can cause mistakes, because a dev could forget to pass it to all withEntities that need it.

The second approach create my own withEntities something like withEntitiesCustomId({idKey, collection?}) this just stores the idKey in the store and then any withEntities* after it can used if present, this is what I'm considering at the moment

**Comment by sasobunny** [comment]
+1 would really need this customized Id property.

I really hope you will add this functionality. Let us know and thank you for your work!

**Comment by gabrielguerrero** [comment]
@sasobunny definitely will support this, there are some new changes coming in ngrx/signals 18 that will help me solve this 
https://github.com/ngrx/platform/pull/4399#discussion_r1651633042

**Comment by gabrielguerrero** [comment]
Good news, I have it worlking, will be comming soon first to the beta channel
PR https://github.com/gabrielguerrero/ngrx-traits/pull/108
Example

```
const entityConfig = {
  entity: type<ProductCustom>(),
  collection: 'products',
  idKey: 'productId' as any,
} as const; // <-- IMPORTANT dont forget to add as const

export const ProductsLocalStore = signalStore(
  { providedIn: 'root' },
  withEntities(entityConfig),
  withCallStatus({ ...entityConfig, initialValue: 'loading' }),
  withEntitiesLocalPagination({
    ...entityConfig,
    pageSize: 5,
  }),
  withEntitiesLocalFilter({
    ...entityConfig,
    defaultFilter: { search: '' },
    filterFn: (entity, filter) =>
      !filter?.search ||
      entity?.name.toLowerCase().includes(filter?.search.toLowerCase()),
  }),
  withEntitiesLocalSort({
    ...entityConfig,
    defaultSort: { field: 'name', direction: 'asc' },
  }),
  withEntitiesSingleSelection({
    ...entityConfig,
  }),
  withEntitiesLoadingCall({
    ...entityConfig,
    fetchEntities: ({ productsFilter }) => {
      return inject(ProductService)
        .getProducts({
          search: productsFilter().search,
        })
        .pipe(
          map((d) =>
            d.resultList.map(({ id, ...product }) => ({
              ...product,
              productId: id,
            })),
          ),
        );
    },
  }),
);
```
Essentially you just need to create entityConfig as shown above and be sure to add as const, and then spread it in each withEntities store feature
There is gonna be a small breaking change when ngrx/signals 18 is released, because it looks like they plan to replace idKey for a function like selectId: (entity) => entity.productId, https://github.com/ngrx/platform/pull/4399#discussion_r1651633042
but Im not sure how long will it take them to release ngrx/signals final 18, it looks like they are planning to new things like support for redux so it could take a while, so I decided it's better to release this soon, the break is very small anyway if it happens in 18

**Comment by gabrielguerrero** [comment]
@jmls @sasobunny  this is now available in 17.9.0-beta.3

**Comment by gabrielguerrero** [comment]
One last comment, like I mentioned there are breaking changes related to custome id in ngrx/signals 18 currently in RC 2, Im already preparing support for it, and changes for ngrx-traits  will be in the beta channel soon in version 18 of this lib, which will require also angular 18 and ngrx/signals 18, 

basically, ngrx/signals 18 replaces idKey for selectId and adds a function  called entityConfig

example of how it will look in version 18 

```typescript
const config = entityConfig({
  entity: type<ProductCustom>(),
  collection: 'products',
  selectId: (entity) => entity.productId, // <--  idKey is replaced by selectId function
});

export const ProductsLocalStore = signalStore(
  { providedIn: 'root' },
  withEntities(config),
  withCallStatus({ ...config, initialValue: 'loading' }),
  withEntitiesLocalPagination({
    ...config,
    pageSize: 5,
  }),
  withEntitiesLocalFilter({
    ...config,
    defaultFilter: { search: '' },
    filterFn: (entity, filter) =>
      !filter?.search ||
      entity?.name.toLowerCase().includes(filter?.search.toLowerCase()),
  }),
  withEntitiesLocalSort({
    ...config,
    defaultSort: { field: 'name', direction: 'asc' },
  }),
  withEntitiesSingleSelection({
    ...config,
  }),
  withEntitiesLoadingCall({
    ...config,
    fetchEntities: ({ productsFilter }) => {
      return inject(ProductService)
        .getProducts({
          search: productsFilter().search,
        })
        .pipe(
          map((d) =>
            d.resultList.map(({ id, ...product }) => ({
              ...product,
              productId: id,
            })),
          ),
        );
    },
  }),
);
```
